src/torchmetrics/detection/mean_ap.py

```diff
@@ -327,15 +387,20 @@ def update(self, preds: List[Dict[str, Tensor]], target: List[Dict[str, Tensor]]
         _input_validator(preds, target, iou_type=self.iou_type)
```

Comment:
Since the iou types are already validated in line 330: ```self.iou_type = _validate_iou_type_arg(iou_type)```, the ```_input_validator``` fn contains redundant validation of iou types. One validation check can be potentially removed.